docs/mvc/tutorials/mvc-with-entity-framework.rst

```diff
@@ -1,8 +1,357 @@
+
+Add a DbContext class
+---------------------
+
+In EF 7, the primary class for interacting with data is ``Microsoft.Data.Entity.DbContext``. Add a class in the *Models* folder that derives from ``DbContext``:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Models/BookContext.cs
+    :language: c#
+
+The ``DbSet`` properties represent collections of entities. These will become tables in the SQL database.
+
+Tip: It’s easy to add the necessary ``using`` directives in Visual Studio. With the insertion point on "BookContext", type ``Ctrl+.`` (Ctrl + period). This will bring up a list of quick action. Select "using ControsoBooks.Models" to add the using directive. 
+
+.. image:: mvc-with-entity-framework/_static/quick-action.png
+
+Next, we'll create some sample data. Add a class named ``SampleData`` in the *Models* folder with the following code:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Models/SampleData.cs
+    :language: c#
+
+You wouldn’t put this into production code, but it’s OK for a sample app. 
+
+Configure Entity Framework
+--------------------------
+
+Open *config.json*. Add the the following entry:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/config.json
+    :language: json
+    :lines: 5-7
+    :dedent: 2
+  
+This defines a connection string to LocalDB, which is a lightweight version of SQL Server Express for development. 
+
+Open the *Startup.cs* file. In the ``ConfigureServices`` method, add:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Startup.cs
+    :language: c#
+    :lines: 32-37
+    :dedent: 12
+
+Add the following code at the end of the *Configure* method:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Startup.cs
+    :language: c#
+    :lines: 72
+    :dedent: 12
+
+Notice in *ConfigureServices* that we call ``Configuration.Get`` to get the database connection string. During development, this setting comes from config.json. When you deploy the app to a production environment, you would store the connection string in an environment variable on the host. 
+
+Here is the complete *Startup.cs* after these changes:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Startup.cs
+    :language: c#
+    :emphasize-lines: 1,5,32-37,71
+
+Use data migrations to create the database
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open *project.json*. In the "commands" section, add an entry for ``EntityFramework.Commands``:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/project.json
+    :language: json
+    :lines: 21-25
+    :dedent: 4
+    :emphasize-lines: 4
+
+Open a command prompt in the project directory (\ContosoBooks\src\ContosoBooks\) and run the following commands:
+
+.. code-block:: none
+
+    dnvm use default
+    dnu restore
+    dnx . ef migration add Initial
+    dnx . ef migration apply
```

Comment:
:cold_sweat: yeah this experience is so ugly... nothing to change in the article (it's spot on)... but we have discussions going on about how to be able to add/apply migrations in ASP.NET 5 without leaving VS (something like the Package Manager Console commands for non-ASP.NET 5 projects).